core/miner/storage_fsm/impl/sealing_impl.cpp

```diff
@@ -1079,40 +1083,34 @@ namespace fc::mining {
     deposit = std::max(deposit, collateral);
 
     logger_->info("submitting precommit for sector: {}", info->sector_number);
-    const auto maybe_signed_msg = api_->MpoolPushMessage(
-        vm::message::UnsignedMessage(
-            miner_address_,
-            minfo.worker,
-            0,
-            deposit,
-            {},
-            {},
-            vm::actor::builtin::v0::miner::PreCommitSector::Number,
-            MethodParams{maybe_params.value()}),
-        api::kPushNoSpec);  // TODO: max fee options
-
-    if (maybe_signed_msg.has_error()) {
-      if (params.replace_capacity) {
-        maybe_error = markForUpgrade(params.replace_sector);
-        if (maybe_error.has_error()) {
-          logger_->error("error re-marking sector {} as for upgrade: {}",
-                         info->sector_number,
-                         maybe_error.error().message());
-        }
-      }
-      logger_->error("pushing message to mpool: {}",
-                     maybe_signed_msg.error().message());
-      FSM_SEND(info, SealingEvent::kSectorChainPreCommitFailed);
-      return outcome::success();
-    }
-
-    std::shared_ptr<SectorPreCommittedContext> context =
-        std::make_shared<SectorPreCommittedContext>();
-    context->precommit_message = maybe_signed_msg.value().getCid();
-    context->precommit_deposit = deposit;
-    context->precommit_info = std::move(params);
-
-    FSM_SEND_CONTEXT(info, SealingEvent::kSectorPreCommitted, context);
+    OUTCOME_TRY(precommit_batcher_->addPreCommit(
+        *info,
+        deposit,
+        params,
+        [=](const outcome::result<CID> &maybe_cid) -> outcome::result<void> {
```

Comment:
`PrecommitCallback` returns `void`, not `outcome::result<void>`

Reply:
fixed